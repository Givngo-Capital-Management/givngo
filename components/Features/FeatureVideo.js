import React from 'react';
import styles from './FeatureVideo.module.scss';

export default function FeatureVideo(props) {
  const { id, title } = props;
  if (!id) {
    return null;
  }
  return (
    <div className={styles['video-container']}>
      <div className={styles.video}>
        <iframe
          title={title}
          src="https://cdn.embedly.com/widgets/media.html?src=https%3A%2F%2Fplayer.vimeo.com%2Fvideo%2F371994595%3Fapp_id%3D122963&dntp=1&url=https%3A%2F%2Fplayer.vimeo.com%2Fvideo%2F371994595&key=96f1f04c5f4143bcb0f2e68c87d65feb&type=text%2Fhtml&schema=vimeo"
          scrolling="no"
          frameBorder="0"
          allow="autoplay; encrypted-media;"
          allowFullScreen
          width="100%"
          height="100%"
          top="0"
          left="0"
        />
      </div>
    </div>
  );
}
