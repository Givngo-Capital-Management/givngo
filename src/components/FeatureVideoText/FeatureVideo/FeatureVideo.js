import React from 'react';
import './FeatureVideo.scss'

const BASE_EMBED_URL = "https://player.vimeo.com/video/";

export default function FeatureVideo(props) {
  const { id, title } = props;
  if (!id) {
    return null;
  }
  const embedUrl = `${BASE_EMBED_URL}${id}?autoplay=1`;
  return (
    <div className="video-container">
      <div className="video">
        <iframe 
          title={title}
          src="https://cdn.embedly.com/widgets/media.html?src=https%3A%2F%2Fplayer.vimeo.com%2Fvideo%2F371994595%3Fapp_id%3D122963&dntp=1&url=https%3A%2F%2Fplayer.vimeo.com%2Fvideo%2F371994595&key=96f1f04c5f4143bcb0f2e68c87d65feb&type=text%2Fhtml&schema=vimeo"
          scrolling="no"
          frameBorder="0"
          allow="autoplay; encrypted-media;"
          allowFullScreen
          width="100%"
          height="100%"
          top="0"
          left="0"
        />
      </div>
    </div>
  )
}