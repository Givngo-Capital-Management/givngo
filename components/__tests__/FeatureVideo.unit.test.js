import React from 'react';
import { shallow } from 'enzyme';
import FeatureVideo from '../Features/FeatureVideo';

describe('FeatureVideo', () => {
  test('renders without props', () => {
    const wrapper = shallow(<FeatureVideo />);
    expect(wrapper).toMatchSnapshot();
  });
  test('render with props', () => {
    const wrapper = shallow(
      <FeatureVideo id="2F371994595" title="Save the Children, Who We Are" />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
