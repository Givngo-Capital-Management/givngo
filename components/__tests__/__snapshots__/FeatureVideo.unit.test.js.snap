// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatureVideo render with props 1`] = `
<div>
  <div>
    <iframe
      allow="autoplay; encrypted-media;"
      allowFullScreen={true}
      frameBorder="0"
      height="100%"
      left="0"
      scrolling="no"
      src="https://cdn.embedly.com/widgets/media.html?src=https%3A%2F%2Fplayer.vimeo.com%2Fvideo%2F371994595%3Fapp_id%3D122963&dntp=1&url=https%3A%2F%2Fplayer.vimeo.com%2Fvideo%2F371994595&key=96f1f04c5f4143bcb0f2e68c87d65feb&type=text%2Fhtml&schema=vimeo"
      title="Save the Children, Who We Are"
      top="0"
      width="100%"
    />
  </div>
</div>
`;

exports[`FeatureVideo renders without props 1`] = `""`;
